src/db.rs

```diff
@@ -607,3 +632,88 @@ impl Database {
         GridFsBucket::new(self.clone(), options.into().unwrap_or_default())
     }
 }
+
+pub(super) struct RunCursorCommand<'conn> {
+    run_command: RunCommand<'conn>,
+    options: RunCursorCommandOptions,
+}
+
+impl<'conn> Operation for RunCursorCommand<'conn> {
+    type O = CursorSpecification;
+    type Command = RawDocumentBuf;
+
+    const NAME: &'static str = "run_cursor_command";
+
+    fn build(&mut self, _description: &StreamDescription) -> Result<Command<Self::Command>> {
+        Operation::build(&mut self.run_command, _description)
```

Comment:
You should be able to call this (and the other places you use `Operation::<foo>`) directly, i.e. `self.run_command.build(_description)`.  What you've got here is the [fully qualified syntax](https://doc.rust-lang.org/reference/expressions/call-expr.html#disambiguating-function-calls), which will always work, but usually isn't needed.